test/sqllogictest/cardinality.slt

```diff
@@ -202,21 +202,21 @@ Explained Query:
           ArrangeBy keys=[[#0]]
             Union
               Negate
-                Project (#0, #1)
-                  Join on=(#2 = least(#0, #1) AND #3 = greatest(#0, #1)) type=differential
-                    ArrangeBy keys=[[greatest(#0, #1), least(#0, #1)]]
-                      Get l0
-                    ArrangeBy keys=[[#1, #0]]
-                      Distinct project=[least(#0, #1), greatest(#0, #1)]
-                        Get l1
-              Get l0
+                CrossJoin type=differential
```

Comment:
Am I reading this correctly in that we're introducing a cross join, but we know that the cross join isn't worse than the join we had before because we apply the filter in `l0`? A short explanation why this isn't a regression would be great!

Reply:
Right, it's not so much introducing a cross join as much as taking the existing join which is already a cross join, and revealing that it is a cross join by proving that the equalities are doing nothing. You can see that `l0` gains a `Distinct project=[]` before the join, which is a 0-ary relation that is either "empty" or "present". It used to contain columns that needed to be equated, but in the new version both `l0` and `l1` have the filter `(3 = least(#1, #2)) AND (4 = greatest(#1, #2)` and the join is equating those two functions in each input (and they are both constants, and equal, so no need to actually form them as a key.

The good news is that the resulting cross join is strictly better than the existing equijoin. Maybe indicates that both "cross join" doesn't need to be terrifying, and the absence of "cross join" doesn't mean we don't have them.

Reply:
A similar form of this would be that
```
SELECT * 
FROM 
    (SELECT * FROM table1 WHERE x = 4) table1, 
    (SELECT * FROM table2 WHERE x = 4) table2
WHERE table1.x = table2.x
```
looooooks like an equijoin, because of `WHERE table1.x = table2.x`, but actually the pre-filtering by `x = 4` ensures that this will be true for 100% of the rows in `table1` and `table2`.

This doesn't happen in MZ at the moment because we recognize `x = 4` as binding a literal to a column, and some transform will pick that up. But this PR is the first time we recognize that expressions other than columns might be literals, like `x + 3 = 4` or `greatest(#0, #1)`.

Reply:
@mgree : @ggevay observes that we should ask you if this change to the test is actually load-bearing: is the modified query plan no longer testing what was intended by the test in the first place?

Reply:
The point of this test was to have the cardinality estimator see a filter with selectivity of 0. It doesn't _seem_ to be making the test defunct. For some reason this test isn't showing the cardinality, though, which we would need to know for sure. 🤔